lib/views/pay/pay_view.dart

```diff
@@ -1,114 +1,255 @@
 import 'package:clnapp/api/api.dart';
 import 'package:clnapp/components/buttons.dart';
+import 'package:clnapp/model/app_model/decode_invoice.dart';
 import 'package:clnapp/model/app_model/pay_invoice.dart';
 import 'package:clnapp/utils/app_provider.dart';
+import 'package:clnapp/views/pay/numberpad_view.dart';
 import 'package:flutter/material.dart';
 import 'package:flutter/services.dart';
 
 class PayView extends StatefulWidget {
   final AppProvider provider;
 
   const PayView({Key? key, required this.provider}) : super(key: key);
+
   @override
   State<PayView> createState() => _PayViewState();
 }
 
 class _PayViewState extends State<PayView> {
-  String state = "";
   String? boltString;
-  int? amountMsat;
-  AppPayInvoice? paymentResponse;
-  String? error;
+  final _invoiceController = TextEditingController();
+  String? display;
+
+  Future<AppPayInvoice> payInvoice(String boltString) async {
+    final response =
+        await widget.provider.get<AppApi>().payInvoice(invoice: boltString);
+    return response;
+  }
 
-  Future<AppPayInvoice> payInvoice(String boltString, int? amountMsat) async {
-    // we don't need error handling here
-    final response = await widget.provider
-        .get<AppApi>()
-        .payInvoice(invoice: boltString, msat: amountMsat);
-    setState(() {
-      state = "Payment Status ${response.payResponse.status}";
-    });
+  Future<AppDecodeInvoice> decodeInvoice(String invoice) async {
+    final response = await widget.provider.get<AppApi>().decodeInvoice(invoice);
     return response;
   }
 
-  Widget _buildMainView() {
+  Widget _buildMainView(BuildContext context) {
+    var size = MediaQuery.of(context).size;
     return Padding(
-      padding: const EdgeInsets.all(30.0),
+      padding: EdgeInsets.symmetric(horizontal: size.width * 0.08),
       child: Column(
         mainAxisAlignment: MainAxisAlignment.spaceEvenly,
         children: [
           TextField(
+            minLines: 5,
+            maxLines: 10,
+            controller: _invoiceController,
             onChanged: (invoiceBolt) {
               boltString = invoiceBolt;
             },
-            decoration: const InputDecoration(
-              border: OutlineInputBorder(),
-              labelText: 'Invoice Bolt11/12',
-              hintText: 'Bolt11/12',
-            ),
-          ),
-          Row(
-            children: [
-              Checkbox(
-                value: amountMsat == null ? false : true,
-                onChanged: (value) {
-                  setState(() {
-                    amountMsat = (value == true ? 0 : null);
-                  });
-                },
+            decoration: InputDecoration(
```

Comment:
We can add the callback onSubmit to the `InputDecoration` that does the same action of the bottom in case the user press enter?

Some docs  https://stackoverflow.com/questions/58320455/how-to-trigger-onsubmitted-on-a-flutter-for-web-textfield